go/vt/vttablet/tabletserver/vstreamer/rowstreamer_test.go

```diff
@@ -33,6 +33,61 @@ import (
 	binlogdatapb "vitess.io/vitess/go/vt/proto/binlogdata"
 )
 
+// TestRowStreamerQuery validates that the correct force index hint and order by is added to the rowstreamer query.
+func TestRowStreamerQuery(t *testing.T) {
+	execStatements(t, []string{
+		"create table t1(id int, uk1 int, val varbinary(128), primary key(id), unique key uk2 (uk1))",
+	})
+	defer execStatements(t, []string{
+		"drop table t1",
+	})
+	engine.se.Reload(context.Background())
+	// We need to StreamRows, to get an initialized RowStreamer.
+	// Note that the query passed into StreamRows is overwritten while running the test.
+	err := engine.StreamRows(context.Background(), "select * from t1", nil, func(rows *binlogdatapb.VStreamRowsResponse) error {
+		type testCase struct {
+			directives      string
+			sendQuerySuffix string
+		}
+		queryTemplate := "select %s id, uk1, val from t1"
+		getQuery := func(directives string) string {
+			if directives == "" {
+				return fmt.Sprintf(queryTemplate, "")
+			}
+			return fmt.Sprintf(queryTemplate, directives)
+		}
+		sendQueryPrefix := "select /*+ MAX_EXECUTION_TIME(3600000) */ id, uk1, val from t1"
+		testCases := []testCase{
+			{"", "force index (`PRIMARY`) order by id"},
+			{"/*vt+ ukColumns=\"uk1\" ukForce=\"uk2\" */", "force index (`uk2`) order by uk1"},
+			{"/*vt+ ukForce=\"uk2\" */", "force index (`uk2`) order by uk1"},
+			{"/*vt+ ukColumns=\"uk1\" */", "order by uk1"},
```

Comment:
I feel like we could probably use a few more test cases, but this might be enough.